src/wrapper.js

```
@@ -0,0 +1,162 @@
+'use strict';
+
+const Promise = require('es6-promise').Promise;
+const assign = require('lodash/assign');
+const find = require('lodash/find');
+const findLast = require('lodash/findLast');
+const sortBy = require('lodash/sortBy');
+
+const wrapper = {};
+const API_LOAD_INTERVAL = 500;
+const API_LOAD_ATTEMPTS = 120;
+let isLoadedPromise;
+
+const StepType = {
+  Step: 0,
+  PopupStep: 1,
+  SuperStep: 2
+};
+
+wrapper.Event = {
+  AfterMenuClose: 'AfterMenuClose',
+  AfterMenuOpen: 'AfterMenuOpen',
+  BBCodeLinkClicked: 'BBCodeLinkClicked',
+  BackButtonClicked: 'BackButtonClicked',
+  BalloonCreated: 'BalloonCreated',
+  BeforeMenuClose: 'BeforeMenuClose',
+  BeforeMenuOpen: 'BeforeMenuOpen',
+  DoneButtonClicked: 'DoneButtonClicked',
+  LauncherClicked: 'LauncherClicked',
+  LauncherIsShown: 'LauncherIsShown',
+  NextButtonClicked: 'NextButtonClicked',
+  NextStepShown: 'NextStepShown',
+  PlayerInitComplete: 'PlayerInitComplete',
+  SurveyPlayed: 'SurveyPlayed',
+  SurveyCompleted: 'SurveyCompleted',
+  TaskCompleted: 'TaskCompleted',
+  TasksLoaded: 'TasksLoaded',
+  UserStoppedWalkthru: 'UserStoppedWalkthru',
+  UserStoppedWalkthruAfterStop: 'UserStoppedWalkthruAfterStop',
+  WalkthruAborted: 'WalkthruAborted',
+  WalkthruFailedToStart: 'WalkthruFailedToStart',
+  WalkthruSelected: 'WalkthruSelected'
+};
+
+let afterApiLoad = function afterApiLoad (fn) {
+  if (!isLoadedPromise) {
+    isLoadedPromise = new Promise(function isLoaded (resolve, reject) {
+      let count = 1;
+      let interval = setInterval(() => {
+        if (typeof WalkMeAPI !== 'undefined') {
+          resolve(true);
+          clearInterval(interval);
+        } else if (count === API_LOAD_ATTEMPTS) {
+          reject(new Error(false));
+          clearInterval(interval);
+        }
+        count++;
+      }, API_LOAD_INTERVAL);
+    });
+  }
+
+  return function isLoaded () {
+    let args = arguments;
+    return isLoadedPromise.then(() => fn.apply(null, args));
```

- [ ] add a error handler for this promise, this should fail silently